src/cpp/src/visual_language/llava_next_video/classes.cpp

```diff
@@ -4,14 +4,171 @@
+
+    // Step 4: (x/255.0 - mean) / std
+    auto result = std::make_shared<v1::Divide>(mean_subtracted, std_const);
+
+    return result;
+}
+
+std::shared_ptr<ov::Node> create_channels_first(std::shared_ptr<ov::Node> input_nhwc) {
+    using namespace ov::op;
+
+    // Transpose from NHWC (0,1,2,3) to NCHW (0,3,1,2)
+    auto transpose_order = v0::Constant::create(ov::element::i64, ov::Shape{4}, std::vector<int64_t>{0, 3, 1, 2});
+    return std::make_shared<v1::Transpose>(input_nhwc, transpose_order);
+}
+
+std::shared_ptr<ov::Node> create_center_crop(std::shared_ptr<ov::Node> input, std::shared_ptr<ov::Node> crop_height, std::shared_ptr<ov::Node> crop_width) {
+    using namespace ov::op;
+
+    // Get input shape
+    auto shape_node = std::make_shared<v3::ShapeOf>(input);
+    auto axis_0 = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{0});
+    auto axis_1 = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{1});
+    auto axis_2 = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{2});
+    auto axis_3 = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{3});
+    auto axis_0_node = v0::Constant::create(ov::element::i64, ov::Shape{}, std::vector<int64_t>{0}); // Gather axis
+
+    auto H = std::make_shared<v8::Gather>(shape_node, axis_1, axis_0_node);
+    auto W = std::make_shared<v8::Gather>(shape_node, axis_2, axis_0_node);
+
+    // Calculate start positions: start_y = (H - crop_height) / 2, start_x = (W - crop_width) / 2
+    auto const_2 = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{2});
+    auto start_y = std::make_shared<v1::Divide>(std::make_shared<v1::Subtract>(H, crop_height), const_2);
+    auto start_x = std::make_shared<v1::Divide>(std::make_shared<v1::Subtract>(W, crop_width), const_2);
+
+    // Calculate end positions: end_y = start_y + crop_height, end_x = start_x + crop_width
+    auto end_y = std::make_shared<v1::Add>(start_y, crop_height);
+    auto end_x = std::make_shared<v1::Add>(start_x, crop_width);
+
+    // Create slice start and stop vectors
+    auto zero = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{0});
+    auto max_val = v0::Constant::create(ov::element::i64, ov::Shape{1}, std::vector<int64_t>{std::numeric_limits<int64_t>::max()});
+
+    // start = [0, start_y, start_x, 0]
+    auto start = std::make_shared<v0::Concat>(ov::NodeVector{zero, start_y, start_x, zero}, 0);
+    // stop = [max, end_y, end_x, max]
+    auto stop = std::make_shared<v0::Concat>(ov::NodeVector{max_val, end_y, end_x, max_val}, 0);
+    // step = [1, 1, 1, 1]
+    auto step = v0::Constant::create(ov::element::i64, ov::Shape{4}, std::vector<int64_t>{1, 1, 1, 1});
+
+    // Apply slice
+    auto sliced = std::make_shared<v8::Slice>(input, start, stop, step);
+
+    return sliced;
+}
+
+std::shared_ptr<ov::Model> create_video_preprocess_model(const ProcessorConfig& config) {
```

Comment:
Can we apply the same OV preprocessing to images as well (in parent class)?

Reply:
preprocessing for images and videos is similar, but the normalization formula and functions used differ and need verification. Once confirmed, they will likely be applied consistently. currently, the focus is on video frame preprocessing, so after validation, this can be reflected in a follow-up PR later